### PR TITLE
Fix portraits/hero information display in battle only mode

### PR DESCRIPTION
**Pull request by idshibanov**
I used battle only mode very frequently for testing.

Morale and luck indicators have image restorer that was overwriting whole display. This caused cases when UI seemed unresponsive (i.e. you change a hero and it still shows the default one).

### CODE REVIEW

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @idshibanov in this case could you please update `HeroesIndicator::SetPos()` method and remove `bool skip_back` input parameter as there are no other cases which call it?

**Comment by ihhub** [comment]
Hi @idshibanov with these changes we have such issue (check second hero moral icon):
image

**Comment by idshibanov** [comment]
> 
> 
> Hi @idshibanov with these changes we have such issue (check second hero moral icon):

Yes, I noticed it too that's why I did not remove `skip_back` parameter initially. We might need it or have to refactor `HeroesIndicator` to fix this as SpriteBack remembers the wrong background as you switch heroes.

**Comment by idshibanov** [comment]
Updated to avoid resetting `SpriteBack` (and calling `SetPos`) on hero change. Had to refactor base class a bit for it to work. Also avoid deleting indicator on every change.

File: src/fheroes2/heroes/heroes_indicator.cpp [hunk 1]
```diff
@@ -91,12 +91,16 @@ const std::string & HeroesIndicator::GetDescriptions( void ) const
     return descriptions;
 }
 
-void HeroesIndicator::SetPos( const Point & pt, bool skip_back )
+void HeroesIndicator::SetHero( const Heroes & h )
+{
+    hero = &h;
```
Comment (ihhub) [inline src/fheroes2/heroes/heroes_indicator.cpp:96]: Hmm... passing a reference and taking a pointer from it is not a very good way to do. We should pass a pointer then.

**Review by ihhub — CHANGES_REQUESTED** [review]
Hi @idshibanov please change input references into pointers so the caller of class will know that this class works with pointers.

**Comment by idshibanov** [comment]
Added additional checks for NULL pointers in classes (since now it's possible to instantiate one without a hero).

**Comment by ihhub** [comment]
@idshibanov thank you very much for fixing this issue!

**Pull request merged**